content/content.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright © 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package containing logic for parsing rule content.
+
+package content
+
+import (
+	"io/ioutil"
+	"path"
+)
+
+// RuleErrorContent wraps content of a single error key.
+type RuleErrorContent struct {
+	Generic  []byte
+	Metadata []byte
+}
+
+// RuleContent wraps all the content available for a rule into a single structure.
+type RuleContent struct {
+	Summary    []byte
```

Comment:
Go-newbie question: why are you using []byte instead strings on this struct? At some point we will be using them as string, so why not to store them as strings from the beginning?

Reply:
@JoseLSegura Will we ever really be using them as strings? The reason was that the conversion from `[]byte` to `string` can fail and I'm not exactly sure what happens when it does. Plus, it seemed like a waste of performance if we didn't really handle them as strings, which we might not.